src/Processor/Braintree/Subscription/Update/BaseUpdateStrategy.php

```diff
@@ -43,9 +44,40 @@ protected function doBraintreeUpdate(Subscription $subscription, ?Plan $newPlan
         $request = Arr::dissoc($request, ['firstBillingDate', 'status']);
         $request['options'] = ['prorateCharges' => true];
 
+        if ($subscription->is(Status::pastDue()) && isset($request['paymentMethodToken'])) {
```

Comment:
Is this happening in the `doBraintreeUpdate` because it's a valid concern in ALL update scenarios?

Could this just as easily be handled in a `pastDue` update strategy?

Reply:
You're saying put it here, essentially?

![Screenshot_20210111_093709](https://user-images.githubusercontent.com/18272064/104195411-97dc8480-53f0-11eb-9c87-e8e5c9534469.png)

Reply:
I don't think so. I am asking if this can be put into it's own class - i.e `class PastDueUpdateStrategy extends BaseUpdateStrategy`.  And then have it resolve separately in the `UpdateStrategyFactory`.

Reply:
Sorry - I was saying do we want to drop that strategy in above, since we know it will fail if it's past due... but I went a more consistent route like what you did with changing billing cycle.

Let me know what you think. I'd like to get this out soon, since it is hurting upgrades and past due people at this point... so we can pair through any changes that we'd need to make from here.

Reply:
Can it be a separate class that is returned by the `UpdateStrategyFactory`? Or is it like the billing cycle strategy in that we need to hear back from their servers before we know?